### PR TITLE
[TS-169] delete dependencies on timeouts

### PR DESCRIPTION
**Pull request by jenchik**

### CODE REVIEW

File: workers/workers_test.go [hunk 1]
```diff
@@ -11,14 +12,47 @@ import (
 )
 
 var (
-	minTickForTest = time.Microsecond * 10
-	uniqWorkerN    = atomic.NewInt32(0)
+	limitTimeForTest = time.Second * 30
```
Comment (pcherednichenko) [inline workers/workers_test.go]: maybe 10 seconds?

File: workers/workers_test.go [hunk 2]
```diff
@@ -11,14 +12,47 @@ import (
...
+	minTickForTest   = time.Microsecond * 10
+	uniqWorkerN      = atomic.NewInt32(0)
 )
 
 func getUniqueWorkerName() string {
 	return fmt.Sprintf("worker %d", uniqWorkerN.Inc())
 }
 
+func tryTestGtZero(cnt *atomic.Int32) bool {
```
Comment (pcherednichenko) [inline workers/workers_test.go]: `checkGtZero(cnt *atomic.Int32) bool {` looks better?

File: workers/workers_test.go [hunk 3]
```diff
@@ -11,14 +12,47 @@ import (
...
+
+		if time.Now().After(limit) {
+			return false
+		}
+
+		runtime.Gosched()
+	}
+}
+
+func tryTestEqual(cnt *atomic.Int32, expected int32) bool {
```
Comment (pcherednichenko) [inline workers/workers_test.go]: `checkEqual(cnt *atomic.Int32, expected int32) bool {` looks better?

File: workers/workers_test.go [hunk 4]
```diff
@@ -63,9 +97,8 @@ func TestWorkerStartAndStop(t *testing.T) {
 			t.FailNow()
 		}
 
-		time.Sleep(minTickForTest * 100)
-		if !assert.True(t, start.Load() > 0, "Cannot start worker") {
-			t.FailNow()
+		if !tryTestGtZero(start) {
+			assert.FailNow(t, "Cannot start worker")
```
Comment (pcherednichenko) [inline workers/workers_test.go]: there is 6 `"Cannot start worker"` errors , maybe move them to var?

Comment (jenchik) [reply workers/workers_test.go]: mb "Cannot start worker" or "Cannot start workers"

File: workers/workers_test.go (hunk 2 above)
Comment (jenchik) [reply workers/workers_test.go]: ok

File: workers/workers_test.go (hunk 3 above)
Comment (jenchik) [reply workers/workers_test.go]: ok

File: workers/workers_test.go (hunk 1 above)
Comment (jenchik) [reply workers/workers_test.go]: to 5s

**Comment by jenchik** [comment]
fixed

File: workers/workers_test.go [hunk 5]
```diff
@@ -32,13 +66,13 @@ func TestWorkerConflictName(t *testing.T) {
 		assert.FailNow(t, "Invalidate workers data, must be 1 worker")
 	}
 
-	// добавляем новый воркер с существующим именем
+	// create new worker with already name
```
Comment (im-kulikov) [inline workers/workers_test.go]: `create new worker with existing name`

Comment (jenchik) [reply workers/workers_test.go]: fixed

**Review by im-kulikov — DISMISSED** [review]
LGTM

**Review by pcherednichenko — DISMISSED** [review]

File: workers/workers_test.go [hunk 6]
```diff
@@ -193,7 +224,7 @@ func TestWorkersWait(t *testing.T) {
 		for i := 0; i < 5; i++ {
 			lockedFlag := false
 			if i == 4 {
-				// только один воркер заблочим
+				// only one worker should be block
```
Comment (kpeo) [inline workers/workers_test.go]: we will block only one worker

Comment (jenchik) [reply workers/workers_test.go]: fixed

**Review by kpeo — APPROVED** [review]

**Review by im-kulikov — APPROVED** [review]
SGTM

**Review by pcherednichenko — APPROVED** [review]

**Pull request merged**